src/devices/Seesaw/SeesawEeprom.cs

```diff
@@ -0,0 +1,80 @@
+        /// <param name="eepromAddress">The point in the EEProm area to write the byte.</param>
+        /// <param name="value">The value to write into the EEProm area.</param>
+        public void WriteEEPromByte(byte eepromAddress, byte value)
+        {
+            WriteEEProm(eepromAddress, new byte[] { value });
+        }
+
+        /// <summary>
+        /// Write a byte array to the EEProm area on the Seesaw module.
+        /// </summary>
+        /// <param name="eepromAddress">The point in the EEProm area to start writing the data.</param>
+        /// <param name="data">The bytes to be written into the EEProm area.</param>
+        public void WriteEEProm(byte eepromAddress, byte[] data)
+        {
+            if (!HasModule(Seesaw_Module.Eeprom))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw EEPROM functionality");
+            }
+
+            Write(Seesaw_Module.Eeprom, (Seesaw_Function)eepromAddress, data);
+        }
+
+        /// <summary>
+        /// Read a byte from the EEProm area on the Seesaw module.
+        /// </summary>
+        /// <param name="eepromAddress">The point in the EEProm area to start reading the data.</param>
+        /// <returns>The data byte read from the EEProm area.</returns>
+        public byte ReadEEPromByte(byte eepromAddress)
+        {
+            if (!HasModule(Seesaw_Module.Eeprom))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw EEPROM functionality");
+            }
+
+            return(ReadByte(Seesaw_Module.Eeprom, (Seesaw_Function)eepromAddress));
+        }
+
+        /// <summary>
+        /// Change the I2C address that the Seesaw board listens on. Note that this will reset communications
+        /// with the host device and dispose the current I2cDevice.
+        /// </summary>
+        /// <param name="i2cAddress">The new I2C address to be used.</param>
+        public void SetI2CAddress(byte i2cAddress)
+        {
+            I2cConnectionSettings oldSsettings = I2cDevice.ConnectionSettings;
+
+            if(i2cAddress != GetI2CAddress())
+            {
+                WriteEEPromByte((byte)Seesaw_Function.Eeprom_I2cAddr, i2cAddress);
+
+                I2cDevice.Dispose();
+
+                Initialize(I2cDevice.Create(new I2cConnectionSettings(oldSsettings.BusId, i2cAddress)));
+            }
+        }
+
+        /// <summary>
+        /// Read the address configured to be used as the I2C address.
+        /// </summary>
+        /// <returns>The data byte representing the I2C address.</returns>
+        public byte GetI2CAddress()
+        {
+            return(ReadEEPromByte((byte)Seesaw_Function.Eeprom_I2cAddr));
```

Comment:
```suggestion
            return ReadEEPromByte((byte)Seesaw_Function.Eeprom_I2cAddr);
```

Reply:
Done in the latest PR update.

Reply:
not seeing this being included

Reply:
I have made this private as it is really only used by another function to determine if the I2C  address has been changed.